### PR TITLE
fix: Split collector into separate module according to manifest type

### PR DESCRIPTION
**Pull request by arajkumar**
- No functional change

### CODE REVIEW

File: src/server.ts
Comment (dgpatelgit) [inline src/server.ts:11]: I think we rename all collector and class with some uniformity like file name will be based on ecosystem and class will be specific to manifest file as below. 
1. File ==> maven.collector.ts, npm.collector.ts, pypi.collector.ts and golang.collector.ts
2. Classes can be ==> pomXmlDependencyCollector, packageJsonDependencyCollector, requirementsTxtDependencyCollector and goModDependencyCollector.

Similarly we can rename the test files as well.

Comment (arajkumar) [reply src/server.ts:11]: How about this?

```
requirments.txt/collector.ts
pipfile/requirments.txt
package.json/collector.ts
yarn.lock/collector.ts
pom.xml/collector.ts
go.mod/collector.ts
```

File: src/collector.ts
Comment (dgpatelgit) [inline src/collector.ts:5]: We can rename this file as "common.ts" and move it under new collector folder, this will ensure all collector code in one directory.

**Review by dgpatelgit — APPROVED** [review]
lgtm, but wait for @dvandra or @mathur07 approval.

File: src/collector.ts
Comment (arajkumar) [reply src/collector.ts:5]: collector.ts is an interface, collector directory implements it for specific file type. usually generic types should stay at top level., not along side with impl.

**Review by mathur07 — APPROVED** [review]

**Pull request merged**
